[14.0] [ADD] Add module for stock report.

feketemihai opened the pull request.

cbssolutions committed:
Merge pull request #1 from OCA/14.0

from oca 14.0 2 ian

cbssolutions committed:
Merge pull request #2 from OCA/14.0

vers from oca 20210204

feketemihai committed:
[14.0] [ADD] Add module for stock report.

dhongu commented:
Acest raport este de fapt o balanda de stoc.

feketemihai commented:
@dhongu Yes, e balanta, plus o sa adaugam in acelasi wizard sa poti printa fisa de magazie pentru un produs sau pentru toate care au avut stocuri/miscari, ca sa avem si un document tipizat.

mariusnick committed:
fix strart date move

mariusnick committed:
add pdf report stock card

Commit:
stock card add test

Commit:
k bugs when select was null

Commit:
show only stocable products

Commit:
also report only for some products

mariusnick committed:
fix commit

mariusnick committed:
fix tests

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+        parent="stock.menu_warehouse_report"
+        groups="account.group_account_user"
+    />
+     <menuitem
```
Un singur meniu e de ajuns, si cele 2 butoane din card report

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+        name="Stock Report"
+        action="action_daily_stock_report"
+        parent="stock.menu_warehouse_report"
+        groups="account.group_account_user"
```
User-ul trebuie sa fie din depozit, nu treuie sa aiba conta ca sa vada rapoartele de stoc

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+        />
+    </record>
+
+<!--    <record id="view_daily_stock_report_line_search" model="ir.ui.view">-->
```
scos ce nu e folosit

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,321 @@
+        required=True,
+    )
+
+    found_product_ids = fields.Many2many(
```
ar trebui 2 campuri, cum era inainte, 1 pentru ce poate user-ul sa faca input, altul daca nu a selectat nici un produs, dupa ce ii dai show card, wizard-ul ramane vizibil, cu man2many tags afiseaza toate produsele din found, daca sunt 200 de produse o sa strice layout-ul si nu mai vezi butoanele
deci product_ids = ce poate user-ul sa faca input; found_product_ids = sa le ia pe toate ce au miscari/cantitati daca user-ul nu selecteaza, vizibil in wizard doar product_ids

mariusnick replied:
ar trebui sa avem cele doua campuri Many2many pe product.product ce am vazut ca nu merge ( imi da eroare) dar pe varianta  acesta functioneza , found_product_ids se populeaza cu ce alege userul si ramane cu produsle selectate sau se pouleaza cu produsele care sunt in  locatie si au miscari( populara este dupa ce se inchide wizardul)

feketemihai replied:
Merge dacă definești tu tabelul de many2many și coloanele pentru fiecare camp

feketemihai commented on l10n_ro_stock_report/README.rst:
```diff
@@ -0,0 +1,107 @@
+======================
```
regenerat readme cu versiunea 14

feketemihai commented on l10n_ro_stock_report/readme/CONTRIBUTORS.rst:
```diff
@@ -0,0 +1,11 @@
+
+  * Fekete Mihai <feketemihai@nexterp.ro>
+  * Alexandru Teodor <teodoralexandru@nexterp.ro>
+  * Musat Natanael <musatnatanaela@nexterp.ro>
```
Musat Natanael -> Musat Natanaela

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,321 @@
+    company_id = fields.Many2one(
+        "res.company", string="Company", default=lambda self: self.env.company
+    )
+    mode = fields.Selection(
```
nu mai trebuie campul

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,321 @@
+
+        for line_report in lines_report:
+            if line_report.data:
+                if line_report.product_id not in self.found_product_ids:
```
ajustat cu product_ids si found_product_ids, sa nu printam fise de magazie goale

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+            </form>
+        </field>
+    </record>
+    <record id="view_stock_card_report_form" model="ir.ui.view">
```
Un singur view, un singur meniu

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+        </field>
+    </record>
+
+     <template id="internal_layout">
```
ar trebui sa folosim layout setat in companie, sa nu facem altul

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+                            INITIAL
+                        </div>
+
+                        <div class="act_as_cell right">
```
ar treui merge-uite sa afisam in ultimele 2 coloane, acum e  un pic ciudat, inital e in coloanele cu intrari si final e sub iesiri...

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,364 @@
+            <div class="act_as_cell left">
+                <t t-esc="line_product.referinta" />
+            </div>
+            <div class="act_as_cell right">
```
ar trebui sa fe formatam cu digits setat in setari la product quantity

feketemihai commented on l10n_ro_stock_report/tests/test_report_stock.py:
```diff
@@ -0,0 +1,155 @@
+                "stock_acc_price_diff": True,
+            }
+        )
+        self.account_difference = self.env["account.account"].search(
```
de vazut in report liniile care nu au fost testate si adaugat si pentru alea

mariusnick committed:
only one menu

mariusnick committed:
product_ids and found_product_ids

Commit:
just some necesary rewrite

mariusnick committed:
bold and italic solduri

mariusnick committed:
remove maintainer-tools

dhongu committed:
fix precommit

dhongu committed:
Merge branch '14.0_add_stock_report' of github.com:NextERP-Romania/l10n-romania into 14.0_add_stock_report

 Conflicts:
	l10n_ro_stock_report/report/stock_report.py
	l10n_ro_stock_report/report/stock_report_view.xml
	l10n_ro_stock_report/static/src/css/report.css
	l10n_ro_stock_report/static/src/js/stock_card_report_backend.js

dhongu committed:
fix precommit

dhongu committed:
separare template, formatare standard, fix camp data

dhongu committed:
fix fields name

dhongu committed:
fix traducere, adaugare grupare dupa produs

mariusnick committed:
add external_layout

mariusnick committed:
fix Romanian

dhongu committed:
adaugare buton de tiparire in raport
eliminare camp found_product_ids
adaugare camp pentru a selecta produsele care au miscari in interval
adaugare raport nou in care nu se genereza cate o pagina pentru fiecare produs

dhongu committed:
utilizare modul web_list_view_general_buttons

dhongu committed:
One product per page

dhongu committed:
adauare web_list_view_general_buttons in test-requirements.txt

dhongu committed:
fix adauare web_list_view_general_buttons in test-requirements.txt

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+
+    products_with_move = fields.Boolean(default=True)
+
+    # found_product_ids = fields.Many2many(
```
Remove if not used

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+    company_id = fields.Many2one(
+        "res.company", string="Company", default=lambda self: self.env.company
+    )
+    # mode = fields.Selection(
```
Remove

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+
+    def _get_report_base_filename(self):
+        self.ensure_one()
+        return "Card %s" % (self.location_id.name)
```
Probably "Stock Sheet %s"

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+
+    def get_products_with_move(self):
+        query = """
+                                SELECT product_id from stock_move as sm
```
Align query to be more readable

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+                "partner_id": row[11],
+            }
+            stock_init += [values]
+        # The records from stock_init are converted for in  stock card. That are, a record  with the stock
```
Use stock sheet instead of stock card

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+
+        lines_report = self.env[line_model].create(sold_stock_init)
+
+        # for line_report in lines_report:
```
Remove since it is not used

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+        found_products = self.product_ids
+        product_list = self.get_products_with_move()
+        found_products |= self.env["product.product"].browse(product_list)
+        # for line in self.line_product_ids:
```
Remove

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+
+        return found_products
+
+    def button_show_card(self):
```
button_show_stock_sheet instead of button_show_card

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+            "l10n_ro_stock_report.action_card_stock_report_line"
+        ).read()[0]
+        action["domain"] = [("report_id", "=", self.id)]
+        action["context"] = {
```
Does it have to be in context or it is get directly from get_general_buttons?

feketemihai commented on l10n_ro_stock_report/report/stock_report.py:
```diff
@@ -0,0 +1,414 @@
+
+    def print_pdf(self):
+        if self.one_product:
+            action_report_stock_card = self.env.ref(
```
Use sheet instead of card

feketemihai commented on l10n_ro_stock_report/report/stock_report_template.xml:
```diff
@@ -0,0 +1,340 @@
+<odoo>
+
+
+    <template id="internal_layout">
```
Remove since it is not used anymore

feketemihai commented on l10n_ro_stock_report/report/stock_report_template.xml:
```diff
@@ -0,0 +1,340 @@
+        </div>
+    </template>
+
+    <template id="report_stock_card">
```
Use sheet instead of card

feketemihai commented on l10n_ro_stock_report/report/stock_report_template.xml:
```diff
@@ -0,0 +1,340 @@
+        </t>
+    </template>
+
+    <template id="report_stock_card_report_base">
```
Use sheet instead of card

feketemihai commented on l10n_ro_stock_report/report/stock_report_template.xml:
```diff
@@ -0,0 +1,340 @@
+            <div class="act_as_row labels">
+                <div class="act_as_cell">Data</div>
+                <div class="act_as_cell">Reference</div>
+                <div class="act_as_cell">Input</div>
```
Should we use Quantity In instead of Input?

feketemihai commented on l10n_ro_stock_report/report/stock_report_view.xml:
```diff
@@ -0,0 +1,156 @@
+                        default_focus="1"
+                        class="oe_highlight"
+                    />
+                    <button
```
Button is not working

feketemihai commented on l10n_ro_stock_report/static/src/js/stock_card_report_button_controller.js:
```diff
@@ -0,0 +1,49 @@
+odoo.define("l10n_ro_stock_report.StockCardController", function (require) {
```
This has to be removed with the buttons module?

feketemihai commented on l10n_ro_stock_report/static/src/js/stock_card_report_button_view.js:
```diff
@@ -0,0 +1,15 @@
+odoo.define("l10n_ro_stock_report.StockCardView", function (require) {
```
This has to be removed

feketemihai commented on l10n_ro_stock_report/static/src/xml/stock_sheet.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Also this one, to be removed, no?

feketemihai commented on l10n_ro_stock_report/views/assets.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
This also has to be removed

mariusnick committed:
replace card with sheet sa

dhongu committed:
eliminare cod JS

dhongu committed:
Merge branch '14.0_add_stock_report' of github.com:NextERP-Romania/l10n-romania into 14.0_add_stock_report

 Conflicts:
	l10n_ro_stock_report/report/stock_report.py

dhongu committed:
fix buton tiparire

dhongu committed:
Coloana de cont si categorie. Afisare in format pivot

dhongu committed:
Returul se trece cu minus in raport

dhongu committed:
Redenumire model, traducere

dhongu committed:
Merge branch '14.0' of github.com:OCA/l10n-romania into 14.0_add_stock_report

dhongu committed:
Optimizare selectie si inserare linii in tabel

dhongu committed:
oprimizare determinare cont

dhongu committed:
fix pre-commit

dhongu committed:
Merge branch '14.0' of github.com:NextERP-Romania/l10n-romania into 14.0_add_stock_report

dhongu committed:
Merge branch '14.0' of github.com:OCA/l10n-romania into 14.0_add_stock_report

feketemihai commented:
Depends on https://github.com/OCA/web/pull/1846

feketemihai committed:
Merge branch '14.0' of github.com:OCA/l10n-romania into 14.0

dhongu committed:
add account_id to svl

dhongu committed:
determinare cont din locatie

dhongu committed:
Merge branch '14.0-add_account_to_svl' of github.com:terrabit-ro/l10n-romania into 14.0_add_stock_report

dhongu committed:
optimizare selectie, selectie cont.

dhongu committed:
fix citire actiune

dhongu committed:
Corectie pentru tratare diferente de fus orar.
